Update agentic search UI for 3.4 (#11584)
opensearch-project/documentation-website: _vector-search/ai-search/building-agentic-search-flows.md

In conversational agents, models are responsible for intelligent reasoning, including tool orchestration, connecting to external sources, and generating appropriate DSL queries. In flow agents, tools run sequentially, and models in the **Query Planning** tool are used only for query generation. Different models are optimized for different scenarios: cost-efficient, fast inference versus resource-intensive, deep-reasoning approaches.

For a list of suggested models compatible with agentic search, see [Model configuration]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/agent-customization/#model-configuration).

### Tools

All agents require the **Query Planning** tool to execute agentic searches. Query generation can be entirely LLM-generated (default) or guided by predefined search templates. Search templates help maintain control over generated queries by directing the model to use known, tested, and performant query patterns.

The following image shows the search templates configuration interface within the **Query Planning** tool.

![Search templates]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-search-template.png)

> Other pre-built tools are available for conversational agents, including **Search Index**, **List Index**, **Index Mapping**, and **Web Search**. Enable these to extend your agent's capabilities. While these represent the most commonly used tools in agentic search, several additional [OpenSearch tools are available]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/tools/index/) and can be configured manually in the **JSON** view.

Reviewer: ```suggestion
Other prebuilt tools are available for conversational agents, including **Search Index**, **List Index**, **Index Mapping**, and **Web Search**. Enable these to extend your agent's capabilities. While these represent the most commonly used tools in agentic search, several additional [OpenSearch tools are available]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/tools/index/) and can be configured manually in the **JSON** view.
```

### MCP servers

> To allow conversational agents to access more external tools, integrate them with MCP servers. To limit the tools that the agents can access, configure filters for each server under **Tool filters**. For more information, see [Using external MCP servers]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/mcp-server).

Reviewer: ```suggestion
To allow conversational agents to access more external tools, integrate them with Model Context Protocol (MCP) servers. To limit the tools that the agents can access, configure filters for each server under **Tool filters**. For more information, see [Using external MCP servers]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/mcp-server).
```

5. Under **Query**, enter a natural language query about your data, as shown in the following image. Select **Search** to run an agentic search.

   ![Agentic search]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-search.png)
6. View the agent's generated query, the search hits, and agent summary.

   The following image shows the generated query DSL that the agent created for the search.

   ![Agentic search query]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-query.png)

   The following image shows the search results with product images and details.

   ![Agentic search hits]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-hits.png)

   The following image shows the agent's step-by-step reasoning and tool usage summary.

   ![Agentic search summary]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-summary.png)

7. Optionally, tune your agent by trying different tools, models, and MCP server integrations to see how the agent performs on different queries against your data.

## Next steps

> - Try agentic search on the [ML Playground](https://ml.playground.opensearch.org/app/opensearch-flow#/workflows/WAmxgJoBWVNV3bhKKnGx?configureAgent=false)

Reviewer: ```suggestion
- Try agentic search on the [ML Playground](https://ml.playground.opensearch.org/app/opensearch-flow#/workflows/WAmxgJoBWVNV3bhKKnGx?configureAgent=false).
```

> - Learn more about [agentic search]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/)

Reviewer: ```suggestion
- Learn more about [agentic search]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/).
```

> - Read the [agentic search blog post](https://opensearch.org/blog/introducing-agentic-search-in-opensearch-transforming-data-interaction-through-natural-language/)

Reviewer: ```suggestion
- Read [this blog post on agentic search](https://opensearch.org/blog/introducing-agentic-search-in-opensearch-transforming-data-interaction-through-natural-language/).
```

## Configuring agents

Agents are very customizable and can be configured in many different ways depending on your target use case. The following image shows a fully-configured conversational agent.

![Agent configuration]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agent-configuration.png)

### Agent types

`Flow` agents are optimized for speed and simplicity in query generation. `Conversational` agents can be configured with more tools and functionality for deeper thinking. Agent types can only be configured **once** during agent creation.

When configuring `Flow` agents, certain models may require manually adding an appropriate `response_filter` in the **Query Planning** tool. For more details, see [Register a flow agent]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/flow-agent/#step-4-register-a-flow-agent). Currently, only `OpenAI` and `Amazon Bedrock Converse` are supported. If you would like native support for other providers, please open a [GitHub issue](https://github.com/opensearch-project/dashboards-flow-framework/issues).
{: .note}

### Models

> In `Conversational` agents, models are responsible for query generation and tool orchestration. In `Flow` agents, models are used strictly for query generation. Additionally, different models are typically optimized for different scenarios: cost-efficient, fast inference versus resource-intensive, deep-thinking approaches.

Reviewer 2: models are responsible for intelligent thinking in terms of picking up the right set of tools, connecting to external sources, searching on web and then finally generating query based on the context gathered?

In flow agents, tools just run sequentially and can be best used to generate queries?

Reviewer 2: Instead of 
```
and models are typically only used for query generation
```
model present in QPT is used for query generation?

Author: agreed. updated

For a list of suggested models compatible with agentic search, see [Model configuration]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/agent-customization/#model-configuration).

### Tools

Agents must have the **Query Planning** tool enabled for executing agentic searches. Query generation can be entirely LLM-generated (default), or can be steered to select from a list of available search templates. Search templates can be useful for maintaining control over the queries generated, forcing the model to leverage known, working, and performant queries.
![Search templates]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-search-template.png)

> Other pre-built tools are available for `Conversational` agents, including **Search Index**, **List Index**, **Index Mapping**, and **Web Search**. Enable these to extend your agent's capabilities.

Reviewer 2: we can link the complete set of built in tools from opensearch here

Author: Listed the ones with first-class support on UI. Good point, will add a line to reference other existing tools
```
While these represent the most commonly used tools in agentic search, several additional [OpenSearch tools are available]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/tools/index/) and can be configured manually using the JSON view.

```

Once the search is completed, under **Generated query**, view the query domain-specific language (DSL) that the agent generated and ran against your cluster. Under **Search results**, view the search response. Depending on the response, you may see different tabs available, such as **Aggregations** if the response contains aggregations, **Visual hits** if the document hits contain images, and **Hits** if the response contains any hits at all. **Raw response** will always be available. If the search was executed with a `Conversational` agent, you can select **View agent summary** to get a step-by-step breakdown of the agent's actions, including the sequence of tools it used and why.

![Agentic search results]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-search-results.png)

### Using in your application

Select **Export** to view all of the underlying resources needed for leveraging agentic search in your downstream application, including the agent and search pipeline details.

![Agentic search export]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-export.png)

> ## Example: Product search with GPT-5

Author: there will be in the blog. prefer to keep the documentation more on the core functionality of the plugin & its features.

   ![Agentic search configuration]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-configuration.png)
4. Under **Agentic search**, select the index you'd like to search against. The agent will already be selected.
5. Under **Query**, enter a natural language query about your data. Select **Search** to run an agentic search.
   ![Agentic search]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-search.png)
6. View the agent's generated query, the search hits, and agent summary.
   ![Agentic search query]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-query.png)

   ![Agentic search hits]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-hits.png)

   ![Agentic search summary]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-example-summary.png)

## Next steps

- Try out Agentic Search right now on the [ML Playground](https://ml.playground.opensearch.org/app/opensearch-flow#/workflows/WAmxgJoBWVNV3bhKKnGx?configureAgent=false)

> - Check out the Agentic Search [documentation]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/) and [blog](https://opensearch.org/blog/introducing-agentic-search-in-opensearch-transforming-data-interaction-through-natural-language/)

Reviewer 2: Should we have blog as another point?

- Check out the OpenSearch agents [documentation]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/agents/index/)

> - Join the discussion on the [OpenSearch forum](https://forum.opensearch.org/)

Reviewer 2: forum link https://forum.opensearch.org/t/use-cases-and-general-feedback-for-agentic-search/27488

### MCP servers

Integrate with MCP servers to let `Conversational` agents access more external tools. Limit which tools the agents can access by configuring filters under **Tool filters** for each server. For more information, see [Using external MCP servers]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/mcp-server).

## Running agentic searches

Test how the agents perform with different indices and different search queries. Collapse the **Configure agent** panel to focus on executing searches and analyzing the results. The following image shows a search for **mens blue shirts** against an index **demo_amazon_fashion** with relevant result images.

![Agent configuration]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-configuration.png)

### Index

Try out different indices in your cluster. Select the **Inspect** button to view the index details. For `Conversational` agents, you can default to **All indices** and let the agent decide.

### Agent

Try out different agents you've created. Select the **Inspect** button to view the agent details.

### Query

> Try out different natural language queries. You can also specify query fields for the agent to focus on in your selected index. To edit the full `agentic` search query directly, toggle to the **JSON** view. For `Conversational` agents, select **Continue conversation** after a search to persist the context for future searches. Select **Clear conversation** to discard any past context and start a new conversation.

Reviewer 2: I meant the field `query_fields`
